subsys/bluetooth/host/classic/avdtp.c

```diff
@@ -340,602 +339,594 @@ static struct bt_avdtp_sep *avdtp_get_sep(uint8_t stream_endpoint_id)
-		uint8_t error_code = 0;
+	int err = 0;
+	int ret;
+	struct bt_avdtp_sep *sep;
+	struct net_buf *rsp_buf;
+	uint8_t error_code = 0;
 
-		sep = avdtp_get_cmd_sep(buf);
-		avdtp_sep_lock(sep);
-		if ((sep == NULL) || (session->ops->suspend_ind == NULL)) {
+	sep = avdtp_get_cmd_sep(buf, &error_code);
+	avdtp_sep_lock(sep);
+	if ((sep == NULL) || (session->ops->suspend_ind == NULL)) {
+		err = -ENOTSUP;
+	} else {
+		if (sep->state != AVDTP_STREAMING) {
 			err = -ENOTSUP;
+			error_code = BT_AVDTP_BAD_STATE;
 		} else {
-			if (sep->state != AVDTP_STREAMING) {
-				err = -ENOTSUP;
-				error_code = BT_AVDTP_BAD_STATE;
-			} else {
-				err = session->ops->suspend_ind(session, sep, &error_code);
-			}
+			err = session->ops->suspend_ind(session, sep, &error_code);
 		}
+	}
 
-		rsp_buf =
-			avdtp_create_reply_pdu(err ? BT_AVDTP_REJECT : BT_AVDTP_ACCEPT,
-					       BT_AVDTP_PACKET_TYPE_SINGLE, BT_AVDTP_SUSPEND, tid);
-		if (!rsp_buf) {
-			avdtp_sep_unlock(sep);
-			return;
-		}
+	rsp_buf = avdtp_create_reply_pdu(err ? BT_AVDTP_REJECT : BT_AVDTP_ACCEPT,
+					 BT_AVDTP_PACKET_TYPE_SINGLE, BT_AVDTP_SUSPEND, tid);
+	if (!rsp_buf) {
+		avdtp_sep_unlock(sep);
+		return;
+	}
 
-		if (err) {
-			if (error_code == 0) {
-				error_code = BT_AVDTP_BAD_ACP_SEID;
-			}
-			LOG_DBG("suspend err code:%d", error_code);
-			net_buf_add_u8(rsp_buf, error_code);
+	if (err) {
+		if (error_code == 0) {
+			error_code = BT_AVDTP_BAD_ACP_SEID;
 		}
+		LOG_DBG("suspend err code:%d", error_code);
+		net_buf_add_u8(rsp_buf, error_code);
+	}
 
-		ret = bt_l2cap_chan_send(&session->br_chan.chan, rsp_buf);
-		if (ret) {
-			net_buf_unref(rsp_buf);
-			LOG_ERR("Error:L2CAP send fail - result = %d", ret);
-		}
-		if (!err && !ret) {
-			bt_avdtp_set_state(sep, AVDTP_OPEN);
-		}
-		avdtp_sep_unlock(sep);
-	} else {
-		struct bt_avdtp_req *req = session->req;
+	ret = bt_l2cap_chan_send(&session->br_chan.chan, rsp_buf);
+	if (ret) {
+		net_buf_unref(rsp_buf);
+		LOG_ERR("Error:L2CAP send fail - result = %d", ret);
+	}
+	if (!err && !ret) {
+		bt_avdtp_set_state(sep, AVDTP_OPEN);
+	}
+	avdtp_sep_unlock(sep);
+}
 
-		if (req == NULL) {
-			return;
-		}
-		k_work_cancel_delayable(&session->timeout_work);
-		if (msg_type == BT_AVDTP_ACCEPT) {
-			bt_avdtp_set_state_lock(CTRL_REQ(req)->sep, AVDTP_OPEN);
-		} else if (msg_type == BT_AVDTP_REJECT) {
-			if (buf->len >= 1U) {
-				uint8_t acp_seid;
-
-				acp_seid = net_buf_pull_u8(buf);
-				if (acp_seid != CTRL_REQ(req)->acp_stream_ep_id) {
-					return;
-				}
-			}
-		}
+static void avdtp_suspend_rsp(struct bt_avdtp *session, struct net_buf *buf, uint8_t msg_type)
+{
+	struct bt_avdtp_req *req = session->req;
+
+	if (req == NULL) {
+		return;
+	}
+	k_work_cancel_delayable(&session->timeout_work);
+	if (msg_type == BT_AVDTP_ACCEPT) {
+		bt_avdtp_set_state_lock(CTRL_REQ(req)->sep, AVDTP_OPEN);
+	} else if (msg_type == BT_AVDTP_REJECT) {
+		avdtp_handle_reject(buf, req);
+	}
+
+	if (req->status == BT_AVDTP_SUCCESS) {
```

Comment:
Here you're doing it right - I just don't understand why you're not consistent about it.